library/psa_crypto.c

```diff
@@ -2735,7 +2735,7 @@ static psa_status_t psa_hmac_setup_internal( psa_hmac_internal_data *hmac,
     status = psa_hash_update( &hmac->hash_ctx, ipad, block_size );
 
 cleanup:
-    mbedtls_platform_zeroize( ipad, key_length );
+    mbedtls_platform_zeroize( ipad, sizeof(ipad) );
```

Comment:
Do we need to fix the memcpy above as well? It also copies to ipad from key, using key_length number of bytes.

Also, how was this overflow caught? Do we need to add any regression test?

Reply:
The memcpy call is only done if `key_length <= block_size` and the function checks that `block_size <= sizeof( ipad )` earlier. So _that_'s ok. However, there are other uses of `key_length` that aren't ok.

Reply:
I caught this using the `psa_crypto_test` on  target that doesn't have support for SHA384 (NRF52480). `psa_hash_setup()` returned an error, going to cleanup.
The `key_length` was 131, and `PSA_HMAC_MAX_HASH_BLOCK_SIZE` is 128